migration_task.go

```diff
@@ -0,0 +1,56 @@
+package migrate
+
+import "github.com/golang-migrate/migrate/v4/database"
+
+// MigrationTask is a callback function type that can be used to execute a
+// Golang based migration step after a SQL based migration step has been
+// executed. The callback function receives the migration and the database
+// driver as arguments.
+type MigrationTask func(migr *Migration, driver database.Driver) error
+
+// options is a set of optional options that can be set when a Migrate instance
+// is created.
+type options struct {
+	// tasks is a map of MigrationTask functions that can be used to execute
+	// a Golang based migration step after a SQL based migration step ha
+	// been executed. The key is the migration version and the value is the
+	// callback function that should be run _after_ the step was executed
+	// (but within the same database transaction).
+	tasks map[uint]MigrationTask
```

Comment:
Just want to get the opinion of reviewers here. Maybe it makes sense to change this, so that it is up to the definition of the code migration itself if it should be re-run on error or not.

I.e. maybe it would actually make sense to make this into:
```
type TaskEntry struct {
    RerunOnError bool
    Task         MigrationTask
}

var tasks map[uint]TaskEntry
```

And where the value of the `RerunOnError` defines if this specific code migration should re-run on the next startup if it errors during the execution of the `MigrationTask` (which should probably be renamed to `CodeMigration`). If the `RerunOnError` is set to `false`, we'll maintain the current behaviour that the database is set to a `dirty` state if the code migration errors.

Reply:
Added as a separate commit in the latest push :)